reconcile/saas_auto_promotions_manager/merge_request_manager/merge_request_manager.py

```diff
@@ -214,8 +215,24 @@ def _unbatch_failed_mrs(self) -> None:
         and close the old batched MR. By doing so, we ensure that unrelated MRs are not blocking each other.
         Unbatched MRs are marked and will never be batched again.
         """
-        # TODO: implemented in follow-up MR
-        pass
+        open_mrs_after_unbatching: list[OpenMergeRequest] = []
+        for mr in self._open_mrs:
+            if not mr.is_batchable:
+                # The MR is already marked as not batchable which means we do not need to unbatch
+                open_mrs_after_unbatching.append(mr)
+                continue
+            if not mr.failed_mr_check:
+                # The MR passed check and doesnt need unbatching
+                open_mrs_after_unbatching.append(mr)
+                continue
+            self._vcs.close_app_interface_mr(
+                mr,
```

Comment:
type mismatch, `close_app_interface_mr` is accepting `ProjectMergeRequest` but passed `OpenMergeRequest`

Reply:
Interesting mypy didnt catch that :thinking: